handlers/issue.go

```diff
@@ -1,71 +1,79 @@
 package handlers
 
 import (
+	"context"
 	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"net/http"
 	"regexp"
 	"strings"
 
 	"github.com/golang/glog"
 	"github.com/google/go-github/github"
 )
 
+const (
+	Assign   = "/assign"
+	Unassign = "/unassign"
+)
+
 type GithubIssue github.Issue
 
 func (s *Server) handleIssueEvent(body []byte) {
-	glog.Infof("Received an Issue Event")
+	glog.Info("Received an Issue Event")
 
 }
 
-func (s *Server) handleIssueCommentEvent(body []byte) {
-	glog.Infof("Received an IssueComment Event")
+//function to handle issue comments
+func (s *Server) handleIssueCommentEvent(body []byte, client *github.Client) {
+	var commentEvent github.IssueCommentEvent
 
-	var prc github.IssueCommentEvent
-	err := json.Unmarshal(body, &prc)
+	err := json.Unmarshal(body, &commentEvent)
 	if err != nil {
 		glog.Errorf("fail to unmarshal: %v", err)
 	}
-	glog.Infof("prc: %v", prc)
-	comment := *prc.Comment.Body
+	//Get the owner info from the req body, and use this owner for the all api's
+	owner := *commentEvent.Repo.Owner.Login
+	ctx := context.Background()
+	//extract the user from login to validate whether user is collaborator or not
+	user := commentEvent.Sender.Login
+	comment := *commentEvent.Comment.Body
+	//split the assignees and operation to be performed.
+	substrings := strings.Split(comment, "@")
 
-	// https://github.com/islinwb/test/pull/1
-	prID := strings.SplitAfter(prc.Issue.PullRequestLinks.GetHTMLURL(), "github.com/")[1]
-	// https://github.com/islinwb/test/pull/1.patch
-	// From <commit ID> MON ...
-	patchURL := prc.Issue.PullRequestLinks.GetPatchURL()
-	resp, err := http.Get(patchURL)
+	assign, _, err := client.Repositories.IsCollaborator(ctx, owner, *commentEvent.Repo.Name, *user)
 	if err != nil {
-		fmt.Println(err)
-	}
+		glog.Fatalf("Not the collaborator: %v", err)
 
-	resp1, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		fmt.Println(err)
 	}
+	reg := regexp.MustCompile("(?mi)^/(un)?assign(( @?[-\\w]+?)*)\\s*$")
+	comm := reg.MatchString(comment)
```

Comment:
Comment please.

Reply:
Should we return immediately when comm is false? And rename comm -> matchAssignOrUnassign?

Good comments and variable names are the important parts of a good code.